lib/cc/analyzer/formatters/html_formatter.rb

```diff
@@ -1,34 +1,280 @@
+                      # scala
+                      "scs" => "scheme",
+                      # scss
+                      "shader" => "glsl",
+                      # sql
+                      "ss" => "scheme",
+                      "st" => "smalltalk",
+                      "styl" => "stylus",
+                      # swift
+                      # tcl
+                      "template" => "json",
+                      "tex" => "latex",
+                      # textile
+                      "tmproj" => "markup",
+                      "tpl" => "smarty",
+                      "ts" => "typescript",
+                      "v" => "verilog",
+                      "vagrantfile" => "ruby",
+                      "vhd" => "vhdl",
+                      # vim
+                      "xaml" => "markup",
+                      "xhtml" => "markup",
+                      "xml" => "markup",
+                      # yaml
+                      "yaws" => "erlang",
+                      "yml" => "yaml",
+                    ).freeze
+
         class ReportTemplate
           include ERB::Util
-          attr_accessor :template, :issues, :issues_by_path
+          attr_reader :issues
 
           TEMPLATE_PATH = File.expand_path(File.join(File.dirname(__FILE__), "templates/html.erb"))
 
-          def initialize(issue_count, issues_by_path)
-            @template = File.read(TEMPLATE_PATH)
-            @issues_by_path = issues_by_path
-            @issue_count = issue_count
+          MARKDOWN_CONFIG = { autolink: true, fenced_code_blocks: true, no_intra_emphasis: true, tables: true }.freeze
+
+          def initialize(issues, filesystem)
+            @issues = issues
+            @filesystem = filesystem
+          end
+
+          def project_name
+            File.basename(filesystem.root)
           end
 
           def render
-            ERB.new(@template).result(binding)
+            template = File.read(TEMPLATE_PATH)
+            ERB.new(template, nil, "-").result(binding)
+          end
+
+          def lang(issue)
+            ext = File.basename(issue["location"]["path"]).split(".").last.downcase
+            Array(LANGUAGES[ext])
+          end
+
+          def body?(issue)
+            issue["content"] && issue["content"]["body"] &&
+              !issue["content"]["body"].strip.empty?
+          end
+
+          def body(issue)
+            return "" unless body?(issue)
+
+            content = issue["content"]["body"].strip
+            markdown(content)
+          end
+
+          def code(path)
+            h filesystem.read_path(path)
+          end
+
+          def location(issue)
+            (lines(issue["location"]["lines"]) ||
+             positions(issue)).uniq.sort.join("-")
+          end
+
+          def other_locations(issue)
+            locations = issue.fetch("other_locations", [])
+            Hash[
+              locations.map do |loc|
+                [loc["path"], lines(loc["lines"]).uniq.sort.join("-")]
```

Comment:
Is this documented anywhere?

I wonder how this is reflected in the UI. Should I just display them one after another in a plain list or is there any hierarchy to it?